src/libutil/experimental-features.cc

```diff
@@ -5,29 +5,131 @@
 
 namespace nix {
 
-std::map<ExperimentalFeature, std::string> stringifiedXpFeatures = {
-    { Xp::CaDerivations, "ca-derivations" },
-    { Xp::ImpureDerivations, "impure-derivations" },
-    { Xp::Flakes, "flakes" },
-    { Xp::NixCommand, "nix-command" },
-    { Xp::RecursiveNix, "recursive-nix" },
-    { Xp::NoUrlLiterals, "no-url-literals" },
-    { Xp::FetchClosure, "fetch-closure" },
-    { Xp::ReplFlake, "repl-flake" },
-    { Xp::AutoAllocateUids, "auto-allocate-uids" },
-    { Xp::Cgroups, "cgroups" },
-    { Xp::DiscardReferences, "discard-references" },
+struct ExperimentalFeatureDetails
+{
+    ExperimentalFeature tag;
+    std::string_view name;
+    std::string_view description;
 };
 
+constexpr std::array<ExperimentalFeatureDetails, 11> xpFeatureDetails = {{
+    {
+        .tag = Xp::CaDerivations,
+        .name = "ca-derivations",
+        .description = R"(
+            Allows derivations to be content-addressed in order to prevent rebuilds
+            when changes to the derivation do not result in changes to the
+            derivation's output. See
+            [__contentAddressed](../language/advanced-attributes.md#adv-attr-__contentAddressed)
+            for more info.
+        )",
+    },
+    {
+        .tag = Xp::ImpureDerivations,
+        .name = "impure-derivations",
+        .description = R"(
+            Allows derivations to produce non-fixed outputs by setting the `__impure`
+            derivation attribute to `true`. See [these release
+            notes](../release-notes/rl-2.8.md) for an example.
+        )",
+    },
+    {
+        .tag = Xp::Flakes,
+        .name = "flakes",
+        .description = R"(
+            Allows for derivations to be packaged in flakes. See the manual entry for
+            [`nix flake`](../command-ref/new-cli/nix3-flake.md) or this [detailed
+            introduction](https://www.tweag.io/blog/2020-05-25-flakes/) for more info.
+        )",
+    },
+    {
+        .tag = Xp::NixCommand,
+        .name = "nix-command",
+        .description = R"(
+            Allows the usage of the new `nix` CLI subcommands, such as `nix build`, `nix
+            develop`, `nix run`, etc. See the manual for
+            [`nix`](../command-ref/new-cli/nix.md) for more info.
+        )",
+    },
+    {
+        .tag = Xp::RecursiveNix,
+        .name = "recursive-nix",
+        .description = R"(
+            Allow Nix derivations to call Nix in order to recursively build derivations.
+            See [this
+            commit](https://github.com/edolstra/nix/commit/1a27aa7d64ffe6fc36cfca4d82bdf51c4d8cf717)
```

Comment:
Please don't link to commit messages. They are inaccessible from the `man` pages, and cannot be maintained. Expand that information here.